src/main/java/serenity/TaskList.java

```diff
@@ -0,0 +1,195 @@
+    /**
+     * Creates task of specified type with given task description.
+     *
+     * @param input Task description specifying type of task.
+     * @return Task of specified type with given task description
+     * @throws SerenityException If type of task is not specified.
+     */
+    public Task createTask(String input) throws SerenityException {
+        Task t;
+
+        if (input.startsWith("todo")) {
+            String[] description = input.split(" ");
+            if (description.length == 1) {
+                throw new SerenityException("Error: The description of a todo cannot be empty.");
+            } else {
+                //remove the type of task
+                String taskDescription = input.split(" ", 2)[1];
+                t = new Todo(taskDescription);
+            }
+        } else if (input.startsWith("deadline")) {
+            String[] description = input.split(" ");
+            if (description.length == 1) {
+                throw new SerenityException("Error: The description of a deadline cannot be empty.");
+            } else {
+                String taskDescription = input.split(" ", 2)[1];
+                String[] parts = taskDescription.split("/by");
+                t = new Deadline(parts[0].strip(), parts[1].strip());
+            }
+        } else if (input.startsWith("event")) {
+            String[] description = input.split(" ");
+            if (description.length == 1) {
+                throw new SerenityException("Error: The description of an event cannot be empty.");
+            } else {
+                String taskDescription = input.split(" ", 2)[1];
+                String[] parts = taskDescription.split("/from");
+                String[] timings = parts[1].split("/to");
+                t = new Event(parts[0].strip(), timings[0].strip(), timings[1].strip());
+            }
+        } else {
+            throw new SerenityException("Error: Type of task is not specified.");
+        }
+
+        return t;
+
+    }
+
+    /**
+     * Adds task to TaskList.
+     *
+     * @param t Task to be added.
+     * @return Message to be displayed when task is added.
+     */
+    public String addTask(Task t) {
+        tasks.add(t);
+        String numOfTasks = tasks.size() == 1 ? "task" : "tasks";
+        return ("Got it. I've added this task:\n" + t
+                + "\nNow you have " + tasks.size() + " " + numOfTasks + " in the list.");
+    }
+
+    /**
+     * Deletes task from TaskList.
+     *
+     * @param input Command containing index of task to delete.
+     * @return Message to be displayed when task is deleted.
+     * @throws SerenityException If task index is not specified.
+     */
+    public String deleteTask(String input) throws SerenityException {
+
+        String[] parts = input.split(" ");
+        if (parts.length == 1) {
+            throw new SerenityException("Error: Missing task index.");
+        }
+
+        int index = Integer.parseInt(input.substring(7)) - 1;
```

Comment:
Just a thought, isn't this going to throw unnecessary exceptions when the user double spaces (either accidentally or intentionally)? eg. Input = `delete  2`